packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -381,6 +470,45 @@ export class ContextsManager {
     });
   }
 
+  public createServiceInformer(
+    kc: KubeConfig,
+    ns: string,
+    context: KubeContext,
+  ): Informer<V1Service> & ObjectCache<V1Service> {
+    const k8sApi = kc.makeApiClient(CoreV1Api);
+    const listFn = (): Promise<{
+      response: IncomingMessage;
+      body: V1ServiceList;
+    }> => k8sApi.listNamespacedService(ns);
+    const path = `/api/v1/namespaces/${ns}/services`;
+    let timer: NodeJS.Timeout | undefined;
+    let connectionDelay: NodeJS.Timeout | undefined;
+    return this.createInformer<V1Service>(kc, context, path, listFn, {
+      resource: 'services',
+      timer: timer,
+      backoff: new Backoff(backoffInitialValue, backoffLimit, backoffJitter),
+      connectionDelay: connectionDelay,
+      onAdd: obj => {
+        this.setStateAndDispatch(context.name, false, { services: true }, state => state.resources.services.push(obj));
+      },
+      onUpdate: obj => {
+        this.setStateAndDispatch(context.name, false, { services: true }, state => {
+          state.resources.services = state.resources.services.filter(o => o.metadata?.uid !== obj.metadata?.uid);
+          state.resources.services.push(obj);
+        });
+      },
+      onDelete: obj => {
+        this.setStateAndDispatch(
+          context.name,
+          false,
+          { services: true },
+          state =>
+            (state.resources.services = state.resources.services.filter(d => d.metadata?.uid !== obj.metadata?.uid)),
+        );
+      },
+    });
+  }
```

Comment:
IMHO this method is almost the same than createDeploymentInformer or createPodInformer except like the path and the listFn

so probably we only need one method to rule them all ?

Reply:
I tried to have only one method, but the differences are in the details, and I'm afraid the result would be unreadable. I would prefer to keep them as is